core/src/main/java/com/digitalpebble/storm/crawler/protocol/httpclient/HttpProtocol.java

```diff
@@ -0,0 +1,198 @@
+
+    private static final org.slf4j.Logger LOG = LoggerFactory
+            .getLogger(HttpProtocol.class);
+
+    private final static PoolingHttpClientConnectionManager CONNECTION_MANAGER = new PoolingHttpClientConnectionManager();
+    static {
+        // Increase max total connection to 200
+        CONNECTION_MANAGER.setMaxTotal(200);
+        // Increase default max connection per route to 20
+        CONNECTION_MANAGER.setDefaultMaxPerRoute(20);
+    }
+
+    private com.digitalpebble.storm.crawler.protocol.HttpRobotRulesParser robots;
+
+    /**
+     * TODO record response time in the meta data, see property
+     * http.store.responsetime.
+     */
+    private boolean responseTime = true;
+
+    // TODO find way of limiting the content fetched
+    private int maxContent;
+
+    private boolean skipRobots = false;
+
+    private HttpClientBuilder builder;
+
+    private RequestConfig requestConfig;
+
+    @Override
+    public void configure(final Config conf) {
+        this.maxContent = ConfUtils.getInt(conf, "http.content.limit",
+                64 * 1024);
+        String userAgent = getAgentString(
+                ConfUtils.getString(conf, "http.agent.name"),
+                ConfUtils.getString(conf, "http.agent.version"),
+                ConfUtils.getString(conf, "http.agent.description"),
+                ConfUtils.getString(conf, "http.agent.url"),
+                ConfUtils.getString(conf, "http.agent.email"));
+
+        this.responseTime = ConfUtils.getBoolean(conf,
+                "http.store.responsetime", true);
+
+        this.skipRobots = ConfUtils.getBoolean(conf, "http.skip.robots", false);
+
+        robots = new HttpRobotRulesParser(conf);
+
+        builder = HttpClients.custom().setUserAgent(userAgent)
+                .setConnectionManager(CONNECTION_MANAGER);
+
+        String proxyHost = ConfUtils.getString(conf, "http.proxy.host", null);
+        int proxyPort = ConfUtils.getInt(conf, "http.proxy.port", 8080);
+
+        boolean useProxy = (proxyHost != null && proxyHost.length() > 0);
+
+        // use a proxy?
+        if (useProxy) {
+            HttpHost proxy = new HttpHost(proxyHost, proxyPort);
+            DefaultProxyRoutePlanner routePlanner = new DefaultProxyRoutePlanner(
+                    proxy);
+            builder.setRoutePlanner(routePlanner);
+        }
+
+        int timeout = ConfUtils.getInt(conf, "http.timeout", 10000);
+        requestConfig = RequestConfig.custom().setSocketTimeout(timeout)
+                .setConnectTimeout(timeout).build();
+    }
+
+    @Override
+    public ProtocolResponse getProtocolOutput(String url, Metadata md)
+            throws Exception {
+
+        LOG.debug("HTTP connection manager stats {}",
+                CONNECTION_MANAGER.getTotalStats());
+
+        // no need to release the connection explicitly as this is handled
+        // automatically
+        CloseableHttpClient httpclient = builder.build();
```

Comment:
If I'm not mistaken, you're confusing the connections and the client. The connection manager manages the connection. But you need to close the client at the end of its use.